multicluster/controllers/multicluster/member/stale_controller.go

```diff
@@ -375,80 +366,156 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(ctx context
 	return nil
 }
 
-func (c *StaleResCleanupController) cleanupMemberClusterAnnounces(ctx context.Context) error {
-	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
-	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+// Run starts the StaleResCleanupController and blocks until stopCh is closed.
+func (c *StaleResCleanupController) Run(stopCh <-chan struct{}) {
+	klog.InfoS("Starting StaleResCleanupController")
+	defer klog.InfoS("Shutting down StaleResCleanupController")
+
+	ctx, _ := wait.ContextForChannel(stopCh)
+
+	go func() {
+		for range c.commonAreaCreationCh {
+			retry.OnError(common.CleanUpRetry, func(err error) bool { return true },
+				func() error {
+					if err := c.cleanUpStaleResources(ctx); err != nil {
+						klog.ErrorS(err, "Failed to clean up stale resources after a ClusterSet is created, will retry later")
+						return err
+					}
+					return nil
+				})
+		}
+	}()
+
+	retry.OnError(common.CleanUpRetry, func(err error) bool { return true },
```

Comment:
Question - why we move this one to be after `c.commonAreaCreationCh`? I feel easier to understand to do cleanup at the very beginning.

Reply:
retry.OnError will block the codes if there is any error. With backoff retry and delay, it means we may miss the channel event because the line `for range` may not started yet.

Reply:
Maybe I can move retry.OnError() on line 389 and wrap it in a go routine before `for range c.commonAreaCreationCh`, then we can remote `<-stopCh` line. It might be easier to understand with `CleanUp` running first. What's your thoughts?

Reply:
In that case, we can start a goroutinue for this cleanup, but do not use goroutinue for `c.commonAreaCreationCh`?

Reply:
Guess we are saying the same thing. That sounds good to me.